rpc/service.go

```diff
@@ -161,7 +162,23 @@ func RunServer(ctx context.Context, srv *grpc.Server, addr string) (WaitFunc, er
 		defer close(rpcWait)
 		defer recovery.LogStackTraceAndContinue("waiting for the rpc service")
 		<-ctx.Done()
-		srv.GracefulStop()
+
+		gracefulStop := make(chan struct{})
+		go func() {
+			defer close(gracefulStop)
+			srv.GracefulStop()
+		}()
+
+		timer := time.NewTimer(2 * time.Minute)
```

Comment:
I think this is a problem https://github.com/10gen/kernel-tools/blob/ad3c80badf498c165c309d50e0a0bc500be36a9d/evergreen/cedar.service#L10. It should probably match Evergreen's https://github.com/10gen/kernel-tools/blob/ad3c80badf498c165c309d50e0a0bc500be36a9d/evergreen/evg-web.service#L10. Could you open a ticket or make that change under this ticket?

Reply:
I just put up a PR! Does this mean that we actually don't need to merge this code, since the timeout will be taken care of by systemd?